File: src/it/java/nl/vollo/kern/test/JsonNodeIsEqualToTextMatcher.java
package nl.vollo.kern.test;

import com.fasterxml.jackson.databind.JsonNode;
import org.hamcrest.BaseMatcher;
import org.hamcrest.Description;
import org.hamcrest.Matcher;

public class JsonNodeIsEqualToTextMatcher extends BaseMatcher<JsonNode> {

    private String expectedValue;

    private JsonNodeIsEqualToTextMatcher(String expectedValue) {
        this.expectedValue = expectedValue;
    }

    public static Matcher<JsonNode> isTextNode(String text) {
        return new JsonNodeIsEqualToTextMatcher(text);
    }

    @Override
    public boolean matches(Object item) {
        if (item instanceof JsonNode) {
            return ((JsonNode) item).asText().equals(expectedValue);
        }
        return false;
    }

    @Override
    public void describeTo(Description description) {
        description.appendValue(expectedValue);
    }
}

File: src/test/kotlin/nl/vollo/kern/test/JsonNodeIsEqualToTextMatcher.kt
package nl.vollo.kern.test

import com.fasterxml.jackson.databind.JsonNode
import org.hamcrest.BaseMatcher
import org.hamcrest.Description

fun isTextNode(text: String): org.hamcrest.Matcher<JsonNode> = JsonNodeIsEqualToTextMatcher(text)

class JsonNodeIsEqualToTextMatcher(private val expectedValue: String): BaseMatcher<JsonNode>() {

    override fun describeTo(description: Description?) {
        description?.appendValue(expectedValue);
    }

    override fun matches(item: Any?): Boolean {
        return if (item != null && item is JsonNode) {
            item.asText() == expectedValue
        } else {
            false
        }
    }

}